### PR TITLE
Support event APIs

### LINKED ISSUE #227: Event API (MQTT)

### PR DESCRIPTION
**Pull request by jkrvivian**
Changes are also made in:
* [inx-mqtt](https://github.com/iotaledger/inx-mqtt/tree/feat/iota-core)
* [inx PR](https://github.com/iotaledger/inx/pull/108)
* [inx-app PR](https://github.com/iotaledger/inx-app/pull/81)

`inx-mqtt` docker image can be packed and successfully tested in docker-network.

Close #227

### CODE REVIEW

File: components/inx/server_commitments.go [hunk 1]
```diff
@@ -46,3 +50,61 @@ func (s *Server) ReadCommitment(_ context.Context, req *inx.CommitmentRequest) (
...
 	return inxCommitment(commitment), nil
 }
+
+func (s *Server) ListenToLatestCommitments(_ *inx.NoParams, srv inx.INX_ListenToLatestCommitmentsServer) error {
+	ctx, cancel := context.WithCancel(Component.Daemon().ContextStopped())
+
+	wp := workerpool.New("ListenToCommitments", workerCount).Start()
+
+	unhook := deps.Protocol.Events.Engine.Notarization.SlotCommitted.Hook(func(commitmentDetails *notarization.SlotCommittedDetails) {
+		inxCommitment := inx.NewCommitmentWithBytes(commitmentDetails.Commitment.ID(), commitmentDetails.Commitment.Data())
```
Comment (muXxer) [inline components/inx/server_commitments.go:60]: Maybe rename to `payload`? :)

File: components/inx/server_commitments.go [hunk 2]
```diff
@@ -46,3 +50,61 @@ func (s *Server) ReadCommitment(_ context.Context, req *inx.CommitmentRequest) (
...
+	// We need to wait until all tasks are done, otherwise we might call
+	// "SendMsg" and "CloseSend" in parallel on the grpc stream, which is
+	// not safe according to the grpc docs.
+	wp.Shutdown()
+	wp.ShutdownComplete.Wait()
+
+	return ctx.Err()
+}
+
+func (s *Server) ListenToFinalizedCommitments(_ *inx.NoParams, srv inx.INX_ListenToFinalizedCommitmentsServer) error {
```
Comment (muXxer) [inline components/inx/server_commitments.go:80]: I'm wondering if we need the possibility to send previous finalized commitments before we actually start sending the new ones. At least we needed that in hornet for the milestones.
There we had this whole `rangeSend` magic (see `ListenToConfirmedMilestones`). Maybe you can have a look and we can discuss if we should add this in a follow up PR?

File: components/inx/server_node.go [hunk 3]
```diff
@@ -11,13 +11,18 @@ import (
...
-		LastConfirmedBlockSlot: uint64(status.LastConfirmedBlockSlot),
-		LatestCommitment:       inxCommitment(status.LatestCommitment),
-		LatestFinalizedSlot:    uint64(status.LatestFinalizedSlot),
-		PruningSlot:            uint64(status.LastPrunedEpoch), // TODO: change name to PruningEpoch
+		IsHealthy:                 status.NodeSynced,
+		LastAcceptedBlockSlot:     uint64(status.LastAcceptedBlockSlot),
+		LastConfirmedBlockSlot:    uint64(status.LastConfirmedBlockSlot),
+		LatestCommitment:          inxCommitment(status.LatestCommitment),
+		LatestFinalizedCommitment: inxCommitment(finalizedCommitment),
+		PruningSlot:               uint64(status.LastPrunedEpoch), // TODO: change name to PruningEpoch
```
[the remark this answers is not in the document]
Comment (jkrvivian) [reply components/inx/server_node.go:25]: That needs an update on inx, too.

File: components/inx/server_utxo.go [hunk 4]
```diff
@@ -164,6 +165,21 @@ func (s *Server) ReadUnspentOutputs(_ *inx.NoParams, srv inx.INX_ReadUnspentOutp
 	return err
 }
 
+func (s *Server) ReadOutputMetadata(_ context.Context, id *inx.OutputId) (*inx.OutputMetadata, error) {
+	outputID := id.Unwrap()
+
+	output, spent, err := deps.Protocol.MainEngineInstance().Ledger.OutputOrSpent(outputID)
```
[the remark this answers is not in the document]
Comment (alexsporn) [reply components/inx/server_utxo.go:171]: The ledgerstate has the index, but when we read this we go through the mempool which has a potentially newer state than the ledgerstate. That is why it is important to compare the slot indexes in the output metadata with the latest commitment of the node, that way you will see if the output was committed or if it is still just in the mempool

File: components/inx/server_node.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexsporn) [reply components/inx/server_node.go:25]: What is the reasoning for adding the finalized commitment here instead of just the index?

**Review by muXxer — APPROVED** [review]

**Review by muXxer — APPROVED** [review]

**Pull request merged**